vault/audited_headers.go

```diff
@@ -120,8 +131,15 @@ func (c *Core) setupAuditedHeadersConfig() error {
 		}
 	}
 
+	// Ensure that we are able to case-sensitively access the headers;
```

Comment:
Removing it will break any headers added in 0.6.5. It only runs during setup, I don't think the expense is concerning.